components/gfx/display_list/mod.rs

```diff
@@ -1043,6 +1043,22 @@ impl ClippingRegion {
     }
 }
 
+impl fmt::Debug for ClippingRegion {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if self.main == max_rect() && self.complex.is_empty() {
```

Comment:
Meh, not really, I think that's fine (it's expected to be debug code anyway). I requested changes just because I can't see how the second branch could be executed, since the condition is the same as in the `if`:

```
if self.main == max_rect() && self.complex.is_empty() {
} else if self.main == max_rect() && self.complex.is_empty() {
}
```